### PR TITLE
Support `use_msi` and `use_cli` authentication switch for AzAPI

### PR DESCRIPTION
**Pull request by ms-zhenhua**
fix https://github.com/Azure/terraform-provider-azapi/issues/203

### CODE REVIEW

**Review by magodo — APPROVED** [review]

**Comment by ms-henglu** [comment]
linked to https://github.com/Azure/terraform-provider-azapi/issues/287

File: internal/provider/provider.go [hunk 1]
```diff
@@ -153,14 +154,23 @@ func azureProvider() *schema.Provider {
 				Description: "Allow OpenID Connect to be used for authentication",
 			},
 
+			// Azure CLI specific fields
+			"use_cli": {
```
Comment (ms-henglu) [inline internal/provider/provider.go:158]: Would you please add the new attributes to the doc here: https://github.com/Azure/terraform-provider-azapi/blob/main/docs/index.md

**Review by ms-henglu — CHANGES_REQUESTED** [review]
Thanks for this PR @ms-zhenhua . I left a minor comment in-line, but once it's fixed up this should be good to go!

**Comment by ms-zhenhua** [comment]
@ms-henglu, thanks for your review. I have updated the doc. Kindly take another review.

**Review by ms-henglu — APPROVED** [review]
LGTM! 💯

**Pull request merged**